docs/json_to_sarif.md

```diff
@@ -0,0 +1,151 @@
+# JSON to SARIF
+
+## Contents
+
+- [SARIF example report](#sarif-example-report)
+- [(Work in Progress) Map](#work-in-progress-map)
+
+## SARIF example report
+
+The following example illustrates how Brakeman should generate a SARIF representation of its analysis.
+
+```json
+{
+  "version": "2.1.0",
+  "$schema": "http://json.schemastore.org/sarif-2.1.0-rtm.4",
+  "runs": [
+    {
+      "tool": {
+        "driver": {
+          "name": "Brakeman",
+          "informationUri": "https://brakemanscanner.org/",
+          "rules": [
+            {
+              "id": 12,
+              "shortDescription": {
+                "text": "Checks for default routes"
+              },
+              "helpUri": "https://brakemanscanner.org/docs/warning_types/default_routes/",
+              "properties": {
+                "warning_type": "Default Routes",
+                "check_name": "DefaultRoutes"
+              }
+            },
+            {
+              "id": 102,
+              "shortDescription": {
+                "text": "Checks for XSS in calls to content_tag"
+              },
+              "helpUri": "https://groups.google.com/d/msg/ruby-security-ann/8B2iV2tPRSE/JkjCJkSoCgAJ",
+              "properties": {
+                "warning_type": "Cross-Site Scripting",
+                "check_name": "ContentTag"
+              }
+            }
+          ]
+        }
+      },
+      "artifacts": [
+        {
+          "location": {
+            "uri": "config/routes.rb"
+          }
+        },
+        {
+          "location": {
+            "uri": "Gemfile.lock"
+          }
+        }
+      ],
+      "results": [
+        {
+          "level": "warning",
+          "message": {
+            "text": "Any public method in `FooPutController` can be used as an action for `put` requests."
```

Comment:
Brakeman warning messages have been (re)designed for translation. It's possible to add another message output format to https://github.com/presidentbeef/brakeman/blob/main/lib/brakeman/messages.rb for SARIF.